avm/res/network/express-route-circuit/main.bicep

```
@@ -0,0 +1,285 @@
+param roleAssignments roleAssignmentType
+
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableTelemetry bool = true
+
+
+var builtInRoleNames = {
+  Contributor: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')
+  'Network Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '4d97b98b-1d4f-4787-a291-c67834d212e7')
+  Owner: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')
+  Reader: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
+  'Role Based Access Control Administrator (Preview)': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'f58310d9-a9f6-439a-9e8d-f62e7b41a168')
+  'User Access Administrator': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '18d7d88d-d35e-4fb5-a5c3-7773c20a72d9')
+}
+
+resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.network-expressroutecircuit.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource expressRouteCircuits 'Microsoft.Network/expressRouteCircuits@2023-04-01' = {
+  name: name
+  location: location
+  tags: tags
+  sku: {
+    name: '${skuTier}_${skuFamily}'
+    tier: skuTier
+    family: skuTier == 'Local' ? 'UnlimitedData' : skuFamily
+  }
+  properties: {
+    allowClassicOperations: allowClassicOperations
+    globalReachEnabled: globalReachEnabled
+    bandwidthInGbps: bandwidthInGbps != 0 ? bandwidthInGbps : null
+    expressRoutePort: !empty(expressRoutePortResourceId) ? {
+      id: expressRoutePortResourceId
+    } : null
+    serviceProviderProperties: {
+      serviceProviderName: serviceProviderName
+      peeringLocation: peeringLocation
+      bandwidthInMbps: bandwidthInMbps
+    }
+    peerings: [
+      {
+        name: peeringType
+        properties: {
+          peeringType: peeringType
+          sharedKey: sharedKey
+          peerASN: peerASN
+          primaryPeerAddressPrefix: primaryPeerAddressPrefix
+          secondaryPeerAddressPrefix: secondaryPeerAddressPrefix
+          vlanId: vlanId
+        }
+      }
+    ]
```

Not exactly what I had in mind. The parameter should still be optional.  I was more thinking along the lines of:

```suggestion
    peerings: peering ? [
      {
        name: peeringType
        properties: {
          peeringType: peeringType
          sharedKey: sharedKey
          peerASN: peerASN
          primaryPeerAddressPrefix: primaryPeerAddressPrefix
          secondaryPeerAddressPrefix: secondaryPeerAddressPrefix
          vlanId: vlanId
        }
      }
    ] : null
```

---

Please add the parameter back and undo the related changes ;)